pallets/evm_account_mapping/src/lib.rs

```diff
@@ -69,10 +71,41 @@ type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Con
 type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
 pub type EIP712ChainID = sp_core::U256;
 pub type EIP712VerifyingContractAddress = sp_core::H160;
+pub type EIP712Signature = [u8; 65];
 
 pub type Nonce = u64;
+pub type AccountId32Bytes = [u8; 32];
+pub type EvmPublicKey = [u8; 33];
 pub type Keccak256Signature = [u8; 32];
 
+pub trait AddressConversion<T: frame_system::Config> {
+	fn try_convert(evm_public_key: &EvmPublicKey) -> Option<T::AccountId>;
+}
+
+pub struct SubstrateAddressConverter<T: frame_system::Config>(PhantomData<T>);
+impl <T> AddressConversion<T> for SubstrateAddressConverter<T>
+	where T: frame_system::Config<AccountId = sp_runtime::AccountId32> {
+	fn try_convert(evm_public_key: &EvmPublicKey) -> Option<T::AccountId> {
+		T::AccountId::decode(&mut &blake2_256(evm_public_key)[..]).ok()
+	}
+}
+
+pub struct EvmTransparentConverter<T: frame_system::Config>(PhantomData<T>);
+impl <T> AddressConversion<T> for EvmTransparentConverter<T>
+	where T: frame_system::Config<AccountId = sp_runtime::AccountId32> {
+	fn try_convert(evm_public_key: &EvmPublicKey) -> Option<T::AccountId> {
+		let h32 = sp_core::H256(sp_core::hashing::keccak_256(evm_public_key));
```

Comment:
The pubkey to be hashed should be uncompressed, see [here](https://github.com/Phala-Network/phala-blockchain/pull/1495/files#diff-e441c91189416dc50daa1df403e57a81fb790923104f16002ba9a198201088d0R179).
We'd better add a unit test to ensure correctness.